plugin.py

```diff
@@ -18,21 +21,29 @@ class LspPyrightPlugin(NpmClientHandler):
     server_directory = "language-server"
     server_binary_path = os.path.join(server_directory, "node_modules", "pyright", "langserver.index.js")
 
-    @classmethod
-    def additional_variables(cls) -> Optional[Dict[str, str]]:
-        variables = super().additional_variables() or {}
-        variables.update(
-            {
-                "sublime_py_files_dir": os.path.dirname(sublime.__file__),
-            }
-        )
-
-        return variables
-
     @classmethod
     def install_in_cache(cls) -> bool:
         return False
 
     @classmethod
     def minimum_node_version(cls) -> Tuple[int, int, int]:
         return (12, 0, 0)
+
+    @classmethod
+    def on_client_configuration_ready(cls, configuration: Dict[str, Any]) -> None:
```

Comment:
Could potentially use `on_settings_read` which provides `sublime.Settings` object. It should be fine to modify it without writing back to it. Then you wouldn't need your custom setting utilities.

(In hindsight, maybe it wasn't necessary to have both)

Reply:
Does it mean I have to do 

```py
    @classmethod
    def on_settings_read(cls, settings: sublime.Settings) -> bool:
        cls.settings = settings

        return False
```

so that I can access settings in `on_client_configuration_ready()`?

Reply:
You can just change configuration from `on_settings_read()`. Just read the original `settings` object, change it and write it back.

Reply:
Resolved in https://github.com/sublimelsp/LSP-pyright/pull/26/commits/b9fb7827e861ad2476b547954a89666b64e12144